dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/RestProtocol.java

```diff
@@ -91,12 +88,11 @@ public int getDefaultPort() {
     protected <T> Runnable doExport(T impl, Class<T> type, URL url) throws RpcException {
         String addr = getAddr(url);
         Class implClass = ApplicationModel.getProviderModel(url.getServiceKey()).getServiceInstance().getClass();
-        RestServer server = servers.get(addr);
-        if (server == null) {
-            server = serverFactory.createServer(url.getParameter(Constants.SERVER_KEY, DEFAULT_SERVER));
-            server.start(url);
-            servers.put(addr, server);
-        }
+        RestServer server = servers.computeIfAbsent(addr, addr0 -> {
```

Comment:
It is recommended to change `addr0` to `restServer`

Reply:
why？？

Reply:
@CrazyHZM please refer to the JavaDoc of `computeIfAbsent`, note the second parameter.

Reply:
Not a grammatical error, just add readability